internal/wrapnix/wrapper.go

```diff
@@ -141,7 +141,11 @@ type createWrapperArgs struct {
 	destPath         string
 }
 
-func createWrapper(args *createWrapperArgs) error {
+func createWrapperIfNeeded(args *createWrapperArgs) error {
+	// If hash is already set, the env is up to date, no need for wrappers.
+	if args.ShellEnvHash == os.Getenv(args.ShellEnvHashKey()) {
+		return nil
+	}
```

Comment:
wait, this change is quite unrelated to this PR. Can we move it out?

I don't think this makes sense. We need the wrappers because the shellenv environment _may_ change after the wrappers have been generated.

Reply:
this is a bad change that snuck in.